### DISCUSSION TITLE
Can Thread nodes discover local IPv4-based DNS services?

### DISCUSSION THREAD

**Discussion by lecontr2** [openthread/openthread#8916] [Q&A] [answered]
I've been attempting to discover services broadcasted by my LG TV and my Roku device but I've been unable to obtain service information using the DNS proxy request through the OTBR. I can see through Wireshark the One-shot Multicast DNS query is executed on behalf of the device, but the node never prints any of the responses on the CLI. 

I can obtain the service information through the Discovery app on iPhone but I've noticed that these services don't provide an IPv6 host, which makes me wonder if this is the reason why the Thread node is unable to find the service. 

I've been using 
`dns browse _airplay._tcp.default.service.arpa.`

**Comment by wgtdkp** [comment, marked as answer]
Yes, the DNS Discovery Proxy "SHOULD" send multicast queries on both IPv4 and IPv6, but it's a bug of OTBR that it's querying on only IPv6.

This should have been fixed by https://github.com/openthread/ot-br-posix/pull/1783.

**Reply by lecontr2** [reply]
Ah I see. Does this have a dependency on the NAT64 feature being enabled on the OTBR for the IPv4 requests to succeed?

**Reply by jwhui** [reply]
Receiving A records via DNS is not dependent on NAT64. However, sending IPv6 packets to IPv4 addresses does require NAT64 functionality.
